View/DataContexts/PermissionAssignmentDataContext.cs

```diff
@@ -0,0 +1,60 @@
+﻿
+using Controller.DbControllers;
+using Model.Annotations;
+using Model.Database.Contexts;
+using Model.DbModels;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Text;
+
+
+namespace View.DataContexts
+{
+    public class PermissionAssignmentDataContext : INotifyPropertyChanged
+    {
+
+        public List<string> Roles => RoleController.Create(DatabaseContext.Instance).GetList().Select(x => x.Designation).ToList();
+        public List<Permission> Permissions => PermissionController.Create(DatabaseContext.Instance).GetList();
+        public Dictionary<Permission, bool[]> PermissionsDict { get; set; }
+
+        private static PermissionAssignmentDataContext instance;
+        public static PermissionAssignmentDataContext Instance
+        {
+            get
+            {
+                if (instance == null)
+                    instance = new PermissionAssignmentDataContext();
+                return instance;
+            }
+        }
+        public PermissionAssignmentDataContext()
+        {
+            PermissionsDict = new Dictionary<Permission, bool[]>();
+            foreach (var rolePermission in RolePermissionsController.Create(DatabaseContext.Instance).GetPermissionsFromRoles(new int[] { 1, 2, 3 })){
```

Comment:
Mocht er rollen toegevoegd worden dan werkt dit niet meer, ik weet niet of dat nu nog uitmaakt aangezien we aan het einde van het project zitten

Reply:
We laten het nu wel zo
maar tip let er vervolgens op als je dynamische data werkt